pkg/containerd/events.go

```diff
@@ -287,6 +401,45 @@ func createPortMappingFromString(portMapping string) (nat.PortMap, error) {
 	return portMap, nil
 }
 
+func extractIPAddress(line string) (string, error) {
+	// Regular expression pattern to match the IP address
+	ipPattern := `inet\s+(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})\/\d{1,2}`
+
+	rx := regexp.MustCompile(ipPattern)
+
+	matches := rx.FindStringSubmatch(line)
+	segments := 2
+	if len(matches) < segments {
+		return "", ErrIPAddressNotFound
+	}
+
+	return matches[1], nil
+}
+
+// ported from:
+// https://github.com/containernetworking/plugins/blob/2b097c5a62dacfd2f9ea4268faa4fc04bd5343f5/pkg/utils/utils.go#L39
+// however module requires Go 1.20
+// MustFormatChainNameWithPrefix generates a chain name similar
+// to FormatChainName, but adds a custom prefix between
+// chainPrefix and unique identifier. Ensures that the
+// generated chain name is exactly maxChainLength chars in length.
+// Panics if the given prefix is too long.
+func mustFormatChainNameWithPrefix(name string, id string, prefix string) string {
+	return mustFormatHashWithPrefix(maxChainLength, chainPrefix+prefix, name+id)
```

Comment:
If we're copying the code (i.e. we can't just upgrade golang), we should skip the extra layer and just hard-code the prefix (since that's always fixed for us).